libpod/runtime_img.go

```diff
@@ -645,36 +645,35 @@ func (r *Runtime) PullImage(imgName string, options CopyOptions) error {
 		// could be trying to pull from registry with short name
 		pullStructs, err = getRegistriesToTry(imgName, r.store, r.config.ImageDefaultTransport)
 		if err != nil {
-			return errors.Wrap(err, "error getting default registries to try")
+			return "", errors.Wrap(err, "error getting default registries to try")
 		}
 	} else {
 		pullStructs, err = r.getPullListFromRef(srcRef, imgName, sc)
 		if err != nil {
-			return errors.Wrapf(err, "error getting pullStruct info to pull image %q", imgName)
+			return "", errors.Wrapf(err, "error getting pullStruct info to pull image %q", imgName)
 		}
 	}
-
 	policyContext, err := getPolicyContext(sc)
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer policyContext.Destroy()
 
 	copyOptions := common.GetCopyOptions(options.Writer, signaturePolicyPath, &options.DockerRegistryOptions, nil, options.SigningOptions, options.AuthFile)
 
 	for _, imageInfo := range pullStructs {
-		if options.Writer != nil {
+		if options.Writer != nil && imageInfo.srcRef.Transport().Name() != DockerArchive && imageInfo.srcRef.Transport().Name() != OCIArchive && imageInfo.srcRef.Transport().Name() != DirTransport {
```

Comment:
OK, might have missed a conversation, but why are these being added?  They don't feel like they're needed to solve the problem this PR is targeted to (Fix misleading print statement)...

Reply:
@TomSweeneyRedHat we only want to print the "Trying to pull..." statement when pulling from a registry. So if the transport is docker-archive, oci-archive, or dir we want to skip the print statement and that is why I added this check.
Initially I was just going to make it print the name of the tar file or directory its pulling from, but kpod load supports piping, e.g "cat file | kpod load", and when we do this we copy the contents to a temp file so doing that would print something along the line "Trying to pull /var/tmp/kpod..", which would confuse the user. That is why I thought of getting rid of that statement completely for those transports.

Reply:
Ok, thanks for the 411.  I still find the code a bit confusing.  I've attempted a suggested rewrite.  Please use it if you think it's better and it works, if not feel free to ignore me!  I think it's a littler clearer to read through, the negatives mislead me first time through.
```
Old:

        for _, imageInfo := range pullStructs {
                if options.Writer != nil && imageInfo.srcRef.Transport().Name() != DockerArchive && imageInfo.srcRef.Transport().Name() != OCIArchive && imageInfo.srcRef.Transport().Name() != DirTransport {
                        io.WriteString(options.Writer, fmt.Sprintf("Trying to pull %s...\n", imageInfo.image))
                }
                if err = cp.Image(policyContext, imageInfo.dstRef, imageInfo.srcRef, copyOptions); err != nil {
                        if options.Writer != nil && imageInfo.srcRef.Transport().Name() != DockerArchive && imageInfo.srcRef.Transport().Name() != OCIArchive && imageInfo.srcRef.Transport().Name() != DirTransport {
                                io.WriteString(options.Writer, "Failed\n")
                        }
                } else {
                        return imageInfo.image, nil
                }
        }
        return "", errors.Wrapf(err, "error pulling image from %q", imgName)

###############

New:

        for _, imageInfo := range pullStructs {
                // Only print header if pulling from a registry.
                if options.Writer == nil ||  imageInfo.srcRef.Transport().Name() == DockerArchive || imageInfo.srcRef.Transport().Name() == OCIArchive || imageInfo.srcRef.Transport().Name() == DirTransport {
                        return imageInfo.image, nil
                }
                io.WriteString(options.Writer, fmt.Sprintf("Trying to pull %s...\n", imageInfo.image))
                }
                if err = cp.Image(policyContext, imageInfo.dstRef, imageInfo.srcRef, copyOptions); err != nil {
                        io.WriteString(options.Writer, "Failed\n")
                } else {
                        return imageInfo.image, nil
                }
        }
        return "", errors.Wrapf(err, "error pulling image from %q", imgName)

```

Reply:
@TomSweeneyRedHat Won't your fail to load the image?  IE if the transport is not pulling from a registry you are not doing cp.Image

Would be better to check the transports that are pulling from a registry, which would future proof this code to not start printing the line if you are not pulling from a registry.

Reply:
I added a comment right before the if statement with all the negatives, hopefully it will be clearer now.